src/common/models/problems.ts

```diff
@@ -94,14 +94,20 @@ class ProblemLocationParams extends Model {
 
 	}
 
-	public get library_id() : number { return this._library_id; }
-	public set library_id(val: string | number) { this._library_id = parseNonNegInt(val);}
+	public get library_id() : number | undefined { return this._library_id; }
+	public set library_id(val: string | number | undefined)
+	{
+		if (val != undefined) this._library_id = parseNonNegInt(val);
+	}
```

Comment:
I do not like functions starting with an open brace on a new line. Personal preference?

Reply:
I don't why I did this.  I don't think the linter picks up on these.